compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -10,25 +10,362 @@ module Language.Bond.Codegen.Java.Class_java
+structFieldDescriptorInitTypeExpr _ BT_Int8 = [lt|com.microsoft.bond.BondTypes.INT8|]
+structFieldDescriptorInitTypeExpr _ BT_Int16 = [lt|com.microsoft.bond.BondTypes.INT16|]
+structFieldDescriptorInitTypeExpr _ BT_Int32 = [lt|com.microsoft.bond.BondTypes.INT32|]
+structFieldDescriptorInitTypeExpr _ BT_Int64 = [lt|com.microsoft.bond.BondTypes.INT64|]
+structFieldDescriptorInitTypeExpr _ BT_UInt8 = [lt|com.microsoft.bond.BondTypes.UINT8|]
+structFieldDescriptorInitTypeExpr _ BT_UInt16 = [lt|com.microsoft.bond.BondTypes.UINT16|]
+structFieldDescriptorInitTypeExpr _ BT_UInt32 = [lt|com.microsoft.bond.BondTypes.UINT32|]
+structFieldDescriptorInitTypeExpr _ BT_UInt64 = [lt|com.microsoft.bond.BondTypes.UINT64|]
+structFieldDescriptorInitTypeExpr _ BT_Float = [lt|com.microsoft.bond.BondTypes.FLOAT|]
+structFieldDescriptorInitTypeExpr _ BT_Double = [lt|com.microsoft.bond.BondTypes.DOUBLE|]
+structFieldDescriptorInitTypeExpr _ BT_Bool = [lt|com.microsoft.bond.BondTypes.BOOL|]
+structFieldDescriptorInitTypeExpr _ BT_String = [lt|com.microsoft.bond.BondTypes.STRING|]
+structFieldDescriptorInitTypeExpr _ BT_WString = [lt|com.microsoft.bond.BondTypes.WSTRING|]
+structFieldDescriptorInitTypeExpr _ BT_Blob = [lt|com.microsoft.bond.BondTypes.BLOB|]
+structFieldDescriptorInitTypeExpr java (BT_Bonded t) = [lt|bondedOf(#{structFieldDescriptorInitTypeExpr java t})|]
+structFieldDescriptorInitTypeExpr java (BT_Nullable t) = [lt|nullableOf(#{structFieldDescriptorInitTypeExpr java t})|]
+structFieldDescriptorInitTypeExpr java (BT_Vector t) = [lt|vectorOf(#{structFieldDescriptorInitTypeExpr java t})|]
+structFieldDescriptorInitTypeExpr java (BT_List t) = [lt|listOf(#{structFieldDescriptorInitTypeExpr java t})|]
+structFieldDescriptorInitTypeExpr java (BT_Set t) = [lt|setOf(#{structFieldDescriptorInitTypeExpr java t})|]
+structFieldDescriptorInitTypeExpr java (BT_Map k v) = [lt|mapOf(#{structFieldDescriptorInitTypeExpr java k}, #{structFieldDescriptorInitTypeExpr java v})|]
+structFieldDescriptorInitTypeExpr _ (BT_TypeParam param) = [lt|#{paramName param}|]
+structFieldDescriptorInitTypeExpr java (BT_UserDefined e@Enum {..} _) = [lt|#{qualifiedDeclaredTypeName java e}.BOND_TYPE|]
+structFieldDescriptorInitTypeExpr java t@(BT_UserDefined s@Struct {..} params) = [lt|#{structFieldDescriptorInitStructExpr java t (qualifiedDeclaredTypeName java s) params}|]
+structFieldDescriptorInitTypeExpr java t@(BT_UserDefined s@Forward {..} params) = [lt|#{structFieldDescriptorInitStructExpr java t (qualifiedDeclaredTypeName java s) params}|]
+structFieldDescriptorInitTypeExpr _ t = error $ "invalid declaration type for structFieldDescriptorInitTypeExpr: " ++ show t
+
+
+-- given struct base type, returns a type descriptor expression
+structBaseDescriptorInitStructExpr :: MappingContext -> Maybe Type -> Text
+structBaseDescriptorInitStructExpr _ Nothing = [lt|null|]
+structBaseDescriptorInitStructExpr java (Just t) = structFieldDescriptorInitTypeExpr java t
```

Comment:
Consider `structBaseDescriptorInitStructExpr java t = maybe [lt|null|] (structFieldDescriptorTypeExpr java) t`

See also: [the `maybe` function](https://hackage.haskell.org/package/base-4.9.1.0/docs/Prelude.html#v:maybe)